# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/shared/PFileSystemManager.ipdl
# Commit: b68e19021833
# Full Hash: b68e1902183357e129a1a53490447fc3cc219f4f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2023-02-07 16:42:31
# Regressor Bug: 1809064
# File Overlap Count: 1
# Description:
#   Bug 1809064 - Add PFileSystemAccessHandleControl top level protocol; r=dom-storage-reviewers,jari
#   
#   Sync access handle methods need to run sync loops to block the thread until a
#   result is available. The main protocol PFileSystemManager and its subprotocol
#   PFileSystemAccessHandle is bound to the main worker event queue, so incoming
# ==============================================================================

diff -r fc3ab5ee8ef3 -r b68e19021833 dom/fs/shared/PFileSystemManager.ipdl
--- a/dom/fs/shared/PFileSystemManager.ipdl	Tue Feb 07 08:21:15 2023 +0000
+++ b/dom/fs/shared/PFileSystemManager.ipdl	Tue Feb 07 08:27:06 2023 +0000
@@ -3,6 +3,7 @@
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 include protocol PFileSystemAccessHandle;
+include protocol PFileSystemAccessHandleControl;
 include protocol PFileSystemWritableFileStream;
 
 include IPCBlob;
@@ -132,6 +133,7 @@
 {
   RandomAccessStreamParams streamParams;
   ManagedEndpoint<PFileSystemAccessHandleChild> accessHandleChildEndpoint;
+  Endpoint<PFileSystemAccessHandleControlChild> accessHandleControlChildEndpoint;
 };
 
 union FileSystemGetAccessHandleResponse